libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -828,6 +867,162 @@ void AP_UAVCAN::notify_state_send()
     _last_notify_state_ms = AP_HAL::native_millis();
 }
 
+void AP_UAVCAN::gnss_send()
+{
+    if (!option_is_set(AP_UAVCAN::Options::SEND_GNSS)) {
+        return;
+    }
+
+    const AP_GPS &gps = AP::gps();
+    const AP_GPS::GPS_Status gps_status = gps.status();
+    const uint32_t now_ms = AP_HAL::native_millis();
+
+    if (_gnss.last_gps_lib_change_ms != gps.last_message_time_ms()) {
+        _gnss.last_gps_lib_change_ms = gps.last_message_time_ms();
+
+        // GPS library just updated, lets send these now
+        _gnss.last_fix2_ms = 0;
+    }
+
+    // send slow, at 5Hz. If there's an update in the GPS library it will trigger faster, typically 10Hz
+    if (!_gnss.last_fix2_ms || (now_ms - _gnss.last_fix2_ms >= 200)) {
```

Comment:
i don't think the 2nd part of this (which seems to try to force 5hz) is right, it should send when we really get a new GPS msg

Reply:
fixed